src/App.tsx

```diff
@@ -1,61 +1,156 @@
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [selectedTitle, setTitle] = useState('');
+  const [selectedUser, setUser] = useState('');
+  const [addNewTodo, setTodo] = useState(todos);
+  const [validationTitle, setValTitle] = useState(true);
+  const [validationUser, setValUser] = useState(true);
+
+  const clearForm = () => {
+    setTitle('');
+    setUser('');
+  };
+
+  useEffect(() => {
+    return () => {
+      return clearForm();
+    };
+  }, [addNewTodo]);
+
+  useEffect(() => {
+    setValTitle(true);
+  }, [selectedTitle]);
+
+  useEffect(() => {
+    setValUser(true);
+  }, [selectedUser]);
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(event) => {
+          event.preventDefault();
+        }}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="titleId">
+            Title:
+          </label>
+          <input
+            type="text"
+            data-cy="titleInput"
+            name="titleId"
+            value={selectedTitle}
+            placeholder="Please enter a title"
+            onChange={event => {
+              const { target } = event;
+
+              setTitle(target.value);
+            }}
+          />
+
+          {!validationTitle
+          && (<span className="error">Please enter a title</span>)}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
+          <label htmlFor="userSelectedId">
+            User:
+          </label>
+
+          <select
+            name="userSelectedId"
+            data-cy="userSelect"
+            value={selectedUser}
+            onChange={event => {
+              const { target } = event;
+
+              setUser(target.value);
+            }}
+          >
+            <option value="" disabled>Choose a user</option>
+
+            {usersFromServer.map(user => (
+              <option
+                key={user.id}
+                value={user.name}
+              >
+                {user.name}
+              </option>
+            ))}
           </select>
 
-          <span className="error">Please choose a user</span>
+          {!validationUser
+          && (<span className="error">Please choose a user</span>)}
         </div>
 
-        <button type="submit" data-cy="submitButton">
+        <button
+          type="submit"
+          data-cy="submitButton"
+          onClick={() => {
+            const findUser = usersFromServer.find(
+              user => user.name === selectedUser,
+            );
+
+            const largeId = [...todos].sort(
+              (a, b) => a.id - b.id,
+            )[todos.length - 1].id;
```

Comment:
something like this is better solution to set an id
const newId = Math.max(...todos.map(todo => todo.id)) + 1;